app/javascript/packs/MainApp/components/RefereeProfileEdit.jsx

```diff
@@ -0,0 +1,259 @@
+import React, { Component, Fragment } from 'react'
+import PropTypes from 'prop-types'
+import axios from 'axios'
+import {
+  Form, Header, Label, Message, Segment
+} from 'semantic-ui-react'
+
+class RefereeProfileEdit extends Component {
+  static propTypes = {
+    defaultValues: PropTypes.shape({
+      firstName: PropTypes.string,
+      lastName: PropTypes.string,
+      bio: PropTypes.string,
+      email: PropTypes.string,
+      showPronouns: PropTypes.bool,
+      nationalGoverningBodies: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.string,
+        name: PropTypes.string
+      }))
+    }).isRequired,
+    onSubmit: PropTypes.func.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    const {
+      defaultValues: {
+        firstName,
+        lastName,
+        bio,
+        email,
+        showPronouns,
+        pronouns,
+        nationalGoverningBodies
+      }
+    } = props;
+
+    this.state = {
+      firstName,
+      lastName,
+      bio,
+      email,
+      showPronouns,
+      pronouns,
+      nationalGoverningBodies,
+      validationErrors: {},
+      availableNationalGoverningBodies: []
+    }
+  }
+
+  componentDidMount() {
+    axios
+      .get('/api/v1/national_governing_bodies')
+      .then(({ data: { data } }) => {
+        this.setState({
+          availableNationalGoverningBodies: data.map(nationalGoverningBody => ({
+            id: nationalGoverningBody.id,
+            name: nationalGoverningBody.attributes.name,
+            website: nationalGoverningBody.attributes.website
+          }))
+        })
+      });
+  }
+
+  getNationalGoverningBodyJsx(nationalGoverningBody) {
+    const { nationalGoverningBodies, validationErrors } = this.state;
+
+    return (
+      <Form.Checkbox
+        id={`nationalGoverningBody[${nationalGoverningBody.id}]`}
+        label={nationalGoverningBody.name}
+        key={nationalGoverningBody.id}
+        checked={nationalGoverningBodies.some(ngb => ngb.id === nationalGoverningBody.id)}
+        onChange={this.changeNationalGoverningBodyCheckbox.bind(this, nationalGoverningBody.id)}
+        error={validationErrors.noNationalGoverningBody}
+      />
+    )
+  }
+
+  changeInput = (event) => {
+    this.setState({
+      [event.target.id]: event.target.value
+    })
+  };
+
+  changeCheckbox = (event) => {
+    this.setState({
+      [event.target.id]: event.target.checked
+    })
+  };
+
+  changeNationalGoverningBodyCheckbox = (id, event) => {
+    const { checked } = event.target;
+
+    this.setState((oldState) => {
+      const {
+        nationalGoverningBodies,
+        availableNationalGoverningBodies
+      } = oldState;
+
+      let newList = nationalGoverningBodies;
+      if (checked) {
+        nationalGoverningBodies.push(
```

Comment:
Looks like this should be newList, we don't want to directly mutate the old state.

Reply:
Good catch! That also means I can’t use `.push()` if `newList` is just a reference.